vmdb/app/controllers/provider_foreman_controller.rb

```diff
@@ -0,0 +1,535 @@
+class ProviderForemanController < ApplicationController
+  before_filter :check_privileges
+  before_filter :get_session_data
+
+  after_filter :cleanup_action
+  after_filter :set_session_data
+
+  def index
+    redirect_to :action => 'explorer'
+  end
+
+  PROVIDER_FOREMAN_X_BUTTON_ALLOWED_ACTIONS = {
+    'provider_foreman_new'     => :provider_foreman_new,
+    'provider_foreman_edit'    => :provider_foreman_edit,
+    'provider_foreman_delete'  => :provider_foreman_delete,
+    'provider_foreman_refresh' => :provider_foreman_refresh,
+  }.freeze
+
+  def x_button
+    @sb[:action] = action = params[:pressed]
+
+    raise ActionController::RoutingError.new('invalid button action') unless
+        PROVIDER_FOREMAN_X_BUTTON_ALLOWED_ACTIONS.key?(action)
+
+    send(PROVIDER_FOREMAN_X_BUTTON_ALLOWED_ACTIONS[action])
+  end
+
+  def provider_foreman_new
+    assert_privileges("provider_foreman_new")
+    @in_a_form = true
+    @provider_foreman = ProviderForeman.new
+    replace_right_cell
+  end
+
+  def provider_foreman_edit
+    assert_privileges("provider_foreman_edit")
+    case params[:button]
+    when "cancel"
+      @in_a_form = false
+      @sb[:action] = nil
+      add_flash(_("%{action} %{model} was cancelled by the user") %
+        {:model  => ui_lookup(:model => "ProviderForeman"),
+         :action => params[:id] == "new" ? "Add of" : "Edit of"})
+      replace_right_cell
+    when "save"
+      if params[:id] == "new"
+        @provider_foreman = ProviderForeman.new(:name    => params[:name],
+                                                :url     => params[:url],
+                                                :zone_id => Zone.find_by_name(MiqServer.my_zone).id)
+        @provider_foreman.update_attributes(:verify_ssl => 1) if params[:verify_ssl] == "on"
```

Comment:
also, pretty sure true/false work here, not just 1/0. Although, not sure what is brought back from the class.

If you need us to store 0/1, let me know and I can test if that would work.